### PR TITLE
Add response read timeouts

### PR DESCRIPTION
**Pull request by pakrym**
Another take on https://github.com/Azure/azure-sdk-for-net/pull/10128 now with the test fixed.

### CODE REVIEW

File: sdk/core/Azure.Core/src/Pipeline/Internal/ResponseBodyPolicy.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        }
+
+        private async Task CopyToAsync(Stream source, Stream destination, CancellationTokenSource cancellationTokenSource)
+        {
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(DefaultCopyBufferSize);
+            try
+            {
+                while (true)
+                {
+                    cancellationTokenSource.CancelAfter(_networkTimeout);
```
Comment (AlexanderSher) [inline sdk/core/Azure.Core/src/Pipeline/Internal/ResponseBodyPolicy.cs]: Shouldn't we set it before loop?

Comment (pakrym) [reply sdk/core/Azure.Core/src/Pipeline/Internal/ResponseBodyPolicy.cs]: It's a per-read timeout so it's set per iteration.

Comment (AlexanderSher) [reply sdk/core/Azure.Core/src/Pipeline/Internal/ResponseBodyPolicy.cs]: Should there be similar checks in `CopyTo`?

Comment (pakrym) [reply sdk/core/Azure.Core/src/Pipeline/Internal/ResponseBodyPolicy.cs]: Stream.Read/Write do not take cancellation tokens anyway so it's impossible to cancel them while they are running.

**Review by tg-msft — APPROVED** [review]

**Pull request merged**